src/main/java/org/testng/ITestResult.java

```diff
@@ -88,4 +91,13 @@
    * @param wasRetried - <code>true</code> if the test was retried and <code>false</code> otherwise.
    */
   void setWasRetried(boolean wasRetried);
+
+  /**
+   * @return - The list of either upstream method(s) or configuration method(s) whose failure led to
+   * the current method being skipped. If there were no such reasons then an empty list is
```

Comment:
Is it possible to have a skipped method without reason?

Reply:
No its not possible. The only time when the return value would be empty is when its queried for a non skipped method. Fixed the javadocs to clarify that.